spec/dsl_spec.rb

```diff
@@ -0,0 +1,29 @@
+require "spec_helper"
+
+RSpec.describe "Pundit RSpec DSL" do
+  before do
+    RSpec.configure do |config|
+      config.include ::Pundit::RSpec::DSL
+    end
```

Comment:
I'm not keen on modifying global state in a `before`-filter. It'll leak to other tests, and we want every test to be isolated from others.

Might I suggest creating a double that mimics rspec behaviour here instead?

```ruby
let(:fake_rspec) do
  double = class_double(RSpec::ExampleGroups)
  double.extend(::Pundit::RSpec::DSL)
  allow(double).to receive(:describe).and_yield
  double
end
```

You would then be able to expect behaviour on the double (note using [yield_control](https://rspec.info/features/3-13/rspec-expectations/built-in-matchers/yield/) matcher here, too):

```ruby
expect { |block| fake_rspec.permissions(..., &block) }.to yield_control
expect(fake_rspec).to have_received(:describe).with(...)
```